lib/matplotlib/backends/backend_webagg_core.py

```diff
@@ -236,8 +236,8 @@ def get_diff_image(self):
                 self._last_renderer, renderer)
             self._force_full = False
             self._png_is_old = False
-
-        return buff
+            self.buff = buff
+        return self.buff
```

Comment:
semantically yes, but the consumer of that return value in `refresh_all` then complains that None is not a buffer. (`send_binary`, that is)

```
    def refresh_all(self):
        if self.web_sockets:
            diff = self.canvas.get_diff_image()
            for s in self.web_sockets:
                s.send_binary(diff)
```